18.md

```diff
@@ -1,19 +1,46 @@
-LUD-18: Proof-of-payer in `payRequest` protocol.
+LUD-18: Payer identity in `payRequest` protocol.
 ================================================
 
 `author: akumaigorodski`
 
 ---
 
-## Adding a `proofofpayer` query parameter in response call
+## Specifying payer identity before sending a payment
 
-This may come especially handy when paying to links extracted from emails since these pay links are very likely to be non-personalized (anyone can obtain lnurl-pay from email as opposed to service providing a personalized pay link to a logged in user).
-
-The `proofofpayer` is an optional ephemeral secp256k1 public key generated by payer, a corresponding private key should be retained by payer, a payee may later ask payer to provide a public key itself or sign a random message using corresponding private key and thus provide a proof of payer.
-
-The callback is thus modified:
+Payer MAY attach a `payerid` query parameter to LNURL-PAY callback with value set to `base64`-encoded JSON array:
 
 ```diff
 - <callback><?|&>amount=<milliSatoshi>
-+ <callback><?|&>amount=<milliSatoshi>&proofofpayer=<hex(ephemeral secp256k1 public key bytes)>
++ <callback><?|&>amount=<milliSatoshi>&payerid=<base64(<array of identity objects>)>
```

Comment:
I suggest relying on "url encoding" here would be better than base64'ing the string, which would also need encoding because of the `=` symbol being a part of the base64 charset.

```JS
encodeURIComponent(<JSON string>)
```